src/buildblock/extend_projdata.cxx

```diff
@@ -145,6 +146,80 @@ extend_segment_in_views(const SegmentBySinogram<float>& sino,
   return out;
 }
 
+/* This function takes symmetries into account to extend segments in any direction.
+   However, it needs testing if it would work for non-direct sinograms.
+*/
+Array<3,float>
+extend_segment(const SegmentBySinogram<float>& segment, const int view_extension,
+               const int axial_extension, const int tangential_extension)
+{
+  Array<3,float> out(segment);
+  BasicCoordinate<3,int> min_dim, max_dim;
+  min_dim[1] = segment.get_min_index() - axial_extension;
+  min_dim[2] = segment[0].get_min_index() - view_extension;
+  min_dim[3] = segment[0][0].get_min_index() - tangential_extension;
+  max_dim[1] = segment.get_max_index() + axial_extension;
+  max_dim[2] = segment[0].get_max_index() + view_extension;
+  max_dim[3] = segment[0][0].get_max_index() + tangential_extension;
+  out.grow(IndexRange<3>(min_dim, max_dim));
+
+  // fill the axial extensions with the same entries from the border
+  for (int axial_edge = 0; axial_edge < axial_extension; axial_edge++)
+  {
+    out[min_dim[1] + axial_edge] = out[min_dim[1] + axial_extension];
+    out[max_dim[1] - axial_edge] = out[max_dim[1] - axial_extension];
+  }
+
+  // fill the view extensions by wrapping around
+  for (int view_edge = 0; view_edge < view_extension; view_edge++)
+  {
+    for (int axial_pos = min_dim[1]; axial_pos <= max_dim[1]; axial_pos++)
+    {
+      // if views cover 360°, we can simply wrap around
+      if (abs(segment.get_proj_data_info_sptr()->get_phi(Bin(0, segment.get_proj_data_info_sptr()->get_max_view_num(), 0, 0)) - 
+              segment.get_proj_data_info_sptr()->get_phi(Bin(0, segment.get_proj_data_info_sptr()->get_min_view_num(), 0, 0))) > _PI)
```

Comment:
This is actually not trivial, because for BlocksOnCylindrical scanners phi doesn't change continuously, so I can't just compare the first and last phi to _PI. Can you think of a check for specifically 180° and 360° that works for both BOC and Cylindrical?

Reply:
> for BlocksOnCylindrical scanners phi doesn't change continuously

I guess you mean it isn't regularly sampled. True. Would something like this work?
```c++
const float phi_diff = max_phi - min_phi;
const float average_phi_sampling = phi_diff/(get_num_views()-1);
const float approx_phi_range = average_phi_sampling * get_num_views();
if (abs(phi_range - 2*_PI) < average_phi_sampling)
...
```
a bit elaborate of course

Reply:
ah well. a factor ~4 off.  I guess this will depend on the size of the gaps etc.

It brings us to an interesting question. The output of `extend_segment` is used as input for interpolation. If the views are very different, then you could argue that we shouldn't be using the "wrap-around" ones anyway.

Reply:
This is true, if the views are very different it would make more sense to just extend with the current view. However, for symmetric BlocksOnCylindrical scanners it would make sense to wrap around, since this would be exactly the same view difference that we also interpolate in other places of the sinograms.